core/infra/repository/world_state_repository_with_level_db.cpp

```diff
@@ -7,82 +7,99 @@
 #include <leveldb/write_batch.h>
 #include <leveldb/db.h>
 
-
+// +------------------------------------------------+
+// | Repository should save string to any database. |
+// +------------------------------------------------+
+// |                                                |
+// | I know ...                                     |
+// |  - leveldb                                     |
+// |                                                |
+// | I don't know                                   |
+// |  - json library                                |
+// |                                                |
+// +------------------------------------------------+
 namespace repository{
 
   // Level DB is known only to me.
   namespace world_state_repository {
 
-      std::unique_ptr<leveldb::DB> db;
-
-      bool loggerStatus(leveldb::Status const status) {
-          if (!status.ok()) {
-              logger::info("merkle_transaction",status.ToString());
-              return false;
+      namespace detail {
+
+              static std::unique_ptr<leveldb::DB> db = nullptr;
+
+              bool loggerStatus(leveldb::Status const status) {
+                  if (!status.ok()) {
+                      logger::info(__FILE__, status.ToString());
+                      return false;
+                  }
+                  return true;
+              }
+
+              void loadDb() {
+                  leveldb::DB *tmpDb;
+                  leveldb::Options options;
+                  options.error_if_exists = false;
+                  options.create_if_missing = true;
+                  loggerStatus(leveldb::DB::Open(options, "/tmp/irohaDB", &tmpDb)); //TODO: This path should be configurable
+                  db.reset(tmpDb);
+              }
           }
-          return true;
-      }
 
-      void loadDb() {
-          leveldb::DB* tmpDb;
-          leveldb::Options options;
-          options.error_if_exists = false;        
-          options.create_if_missing = true;
-          loggerStatus(leveldb::DB::Open(options, "/tmp/irohaDB", &tmpDb)); //TODO: This path should be configurable
-          db.reset(tmpDb);
-      }
+      bool add(const std::string &key, const std::string &value) {
+          if (detail::db == nullptr) detail::loadDb();
 
-      bool add(const std::string& key, const std::string& value){
-          return loggerStatus( db->Put(leveldb::WriteOptions(), key, value));
+          return detail::loggerStatus(detail::db->Put(leveldb::WriteOptions(), key, value));
       }
 
-      bool update(const std::string& key, const std::string& value){
+      bool update(const std::string &key, const std::string &value) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The function 'update' is never used.](https://www.codacy.com/app/soramitsu/iroha/file/4256547628/issues/source?bid=3712501&fileBranchId=3831158#l54)